crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -6338,12 +6356,19 @@ impl<'db> TypeInferenceBuilder<'db> {
                         }
                     }
 
-                    report_non_subscriptable(
-                        &self.context,
-                        value_node.into(),
-                        value_ty,
-                        "__class_getitem__",
-                    );
+                    // TODO: properly handle old-style generics; get rid of this temporary hack
+                    if !value_ty.into_class_literal().is_some_and(|class| {
+                        class
+                            .iter_mro(self.db(), None)
+                            .any(|base| base == ClassBase::Dynamic(DynamicType::SubscriptedGeneric))
```

Comment:
nit: Does it work to change this to the following? (If so, I'm surprised clippy didn't warn about that, I'm pretty sure it's suggested that kind of change to me before)

```suggestion
                        class
                            .iter_mro(self.db(), None)
                            .contains(&ClassBase::Dynamic(DynamicType::SubscriptedGeneric))
```

(I was also going to suggest `class.is_subclass_of(...)` at first, but that takes in a `ClassType`, not a `ClassBase`.  Maybe change that method to take in `impl Into<ClassBase>`? That should still support all of the existing call sites, and you let you use it here too.)

Reply:
I think the `.contains()` requires the third-party `Itertools` trait (which we already have as a dependency, but may not be imported in this module yet). I'm happy to use it here since we already have it as a dependency, just clarifying why Clippy might not be suggesting it 😄

Reply:
Ah no it's because it was on a `Vec`, not an `Iterator`!